tensorboard/plugins/hparams/backend_context.py

```diff
@@ -136,10 +138,25 @@ def read_scalars(self, experiment_id, run, tag):
         Returns:
           A list of `plugin_event_accumulator.TensorEvent` values.
         """
-        assert isinstance(experiment_id, str), (
+        data_provider_output = self._tb_context.data_provider.read_scalars(
             experiment_id,
-            type(experiment_id),
+            plugin_name=scalar_metadata.PLUGIN_NAME,
+            run_tag_filter=provider.RunTagFilter([run], [tag]),
+            downsample=(self._tb_context.sampling_hints or {}).get(
+                scalar_metadata.PLUGIN_NAME, 1000
+            ),
         )
+        data = data_provider_output.get(run, {}).get(tag)
+        if data is None:
+            raise KeyError("No scalar data for run=%r, tag=%r" % (run, tag))
+        return [
+            event_accumulator.TensorEvent(
```

Comment:
Can we maybe file a cleanup github issue to remove the vestigial bits of the multiplexer API from the hparams plugin and/or from plugins generally after they're migrated?  Would be nice to at least leave a TODO here, since it's kind of weird to still have TensorEvent used for this.

Reply:
Yes, definitely. I was planning to do so right after these changes
(switch these functions to directly return the data provider results and
simultaneously migrate callers), but wanted to get these in first to
unblock some internal work. Filed #3425.

For what it’s worth, this is not a performance bottleneck. The
post-processing takes on the order of 1–5 microseconds. The call to
`data_provider.read_scalars` is slow because it calls `list_scalars`
internally, and `list_scalars` itself iterates over all the tensors to
generate metadata that we just throw away. It currently takes on the
order of 10 milliseconds, so that’s not good. And the hparams usage of
`read_scalars` is slow, too, because we call it once for each of the
O(|metrics| × |sessions|) time series instead of passing a single
`RunTagFilter` that represents the cross product, so that’s not good,
either. The data-munging here is negligible in comparison.

(I will see if I can speed up `_list_scalars` before merging this PR.)

Reply:
Yeah, I wasn't concerned at all about performance.  Just seemed like a vestigial weirdness to leave behind - thanks for filing the issue!